src/fireedge/src/modules/containers/VirtualNetworks/VirtualNetworks.js

```diff
@@ -0,0 +1,214 @@
+ *
+ * @returns {ReactElement} Virtual Networks list and selected row(s)
+ */
+export function VirtualNetworks() {
+  const [selectedRows, setSelectedRows] = useState(() => [])
+  const actions = VnsTable.Actions()
+  const { zone } = useGeneral()
+
+  return (
+    <TranslateProvider>
+      <ResourcesBackButton
+        selectedRows={selectedRows}
+        setSelectedRows={setSelectedRows}
+        useUpdateMutation={VnAPI.useUpdateVNetMutation}
+        zone={zone}
+        actions={actions}
+        table={(props) => (
+          <VnsTable.Table
+            onSelectedRowsChange={props.setSelectedRows}
+            globalActions={props.actions}
+            useUpdateMutation={props.useUpdateMutation}
+            onRowClick={props.resourcesBackButtonClick}
+            zoneId={props.zone}
+            initialState={{
+              selectedRowIds: props.selectedRowsTable,
+            }}
+          />
+        )}
+        simpleGroupsTags={(props) => (
+          <GroupedTags
+            tags={props.selectedRows}
+            handleElement={props.handleElement}
+            onDelete={props.handleUnselectRow}
+          />
+        )}
+        info={(props) => {
+          const propsInfo = {
+            vnet: props?.selectedRows?.[0]?.original,
+            selectedRows: props?.selectedRows,
+          }
+          props?.gotoPage && (propsInfo.gotoPage = props.gotoPage)
+          props?.unselect && (propsInfo.unselect = props.unselect)
+
+          return <InfoTabs {...propsInfo} />
+        }}
+      />
+    </TranslateProvider>
+  )
+}
+
+/**
+ * Displays details of a Virtual Network.
+ *
+ * @param {VirtualNetwork} vnet - Virtual Network to display
+ * @param {Function} [gotoPage] - Function to navigate to a page of a Virtual Network
+ * @param {Function} [unselect] - Function to unselect
+ * @returns {ReactElement} Virtual Network details
+ */
+const InfoTabs = memo(({ vnet, gotoPage, unselect }) => {
+  const [get, { data: lazyData, isFetching }] = VnAPI.useLazyGetVNetworkQuery()
+  const id = vnet?.ID ?? lazyData?.ID
+
+  const { settings: { FIREEDGE: fireedge = {} } = {} } = useAuth()
+  const { FULL_SCREEN_INFO } = fireedge
+  const { fullViewMode } = SERVER_CONFIG
+  const fullModeDefault =
+    FULL_SCREEN_INFO !== undefined ? FULL_SCREEN_INFO === 'true' : fullViewMode
+  const { isFullMode } = useGeneral()
+  const { setFullMode } = useGeneralApi()
+
+  useEffect(() => {
+    !isFullMode && gotoPage()
+  }, [])
+
+  return (
+    <Stack overflow="auto">
+      <Stack
+        direction="row"
+        alignItems="center"
+        justifyContent="space-between"
+        gap={1}
+        mx={1}
+        mb={1}
+      >
+        <Stack direction="row">
+          {fullModeDefault && (
+            <SubmitButton
+              data-cy="detail-back"
+              icon={<NavArrowLeft />}
+              tooltip={Tr(T.Back)}
+              isSubmitting={isFetching}
+              onClick={() => unselect()}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using Function.prototype.bind and arrows functions as attributes will negatively impact performance in React. Each time the parent is rendered, the function will be re-created and trigger a render of the component causing excessive renders and more memory use.